akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -164,8 +162,9 @@ final class ORMap[A <: ReplicatedData] private[akka] (
               s"[${thisValue.getClass.getName}], got [${thatValue.getClass.getName}]"
             throw new IllegalArgumentException(errMsg)
           }
+          val mergedValue = thisValue.merge(thatValue)
           // TODO can we get rid of these (safe) casts?
-          val mergedValue = thisValue.merge(thatValue.asInstanceOf[thisValue.T]).asInstanceOf[A]
+          //          val mergedValue = thisValue.merge(thatValue.asInstanceOf[thisValue.T]).asInstanceOf[A]
```

Comment:
This is the problem that I'm stuck on. Merging the values in ORMap. 

    type mismatch; found : thatValue.type (with underlying type A) required: _$6